test/README.md

```diff
@@ -32,19 +32,42 @@ Example command:
 werft job run github -j .werft/run-integration-tests.yaml -a namespace=staging-gpl-2658-int-tests -a version=gpl-2658-int-tests.57 -f
 ```
 
-### Manually against a Kubernetes cluster
+### Manually
 
-You may want to run tests to assert whether your Gitpod installation is successfully integrated.
+You may want to run tests to assert whether a Gitpod installation is successfully integrated.
 
-To test your Gitpod installation:
+#### Using a pod
 
-1. Set your kubectl context to the cluster you want to test
-2. Integrate the Gitpod installation with OAuth for Github and/or Gitlab, otherwise related tests may fail
-3. Clone this repo, and `cd` to `./gitpod/test`
-4. Run the tests like so
+Best for when you want to validate an environment.
+
+1. Update image name in `integration.yaml` for job `integration-job` to latest built by werft.
+2. Optionally add your username in that job argument or any other additional params.
+2. Apply yaml file that will add all necessary permissions and create a job that will run tests.
+   * [`kubectl apply -f ./integration.yaml`](./integration.yaml)
+3. Check logs to inspect test results like so `kubectl logs -f jobs/integration-job`.
+4. Tear down the integration user and job when testing is done.
+   * [`kubectl delete -f ./integration.yaml`](./integration.yaml)
+
+#### Go test
+
+Best for when you're actively developing Gitpod.
+Test will work if images that they use are already cached by gitpod instnance. If not, they might fail if it takes too long to pull an image.
+There are 4 different types of tests:
+1. Enterprise specific, that require valid license to be installed. Run those with `-enterprise=true`
+2. Tests that require correct user (user should have github OAuth integration setup with gitpod). Run those with `-username=<gitpod_username>`. Make sure to load https://github.com/gitpod-io/gitpod-test-repo and https://github.com/gitpod-io/gitpod in gitpod env that you are testing to verify that it is able to pull an image. Wait for it to finish pulling gitpod image, this will ensure that test will not fail due to timeout while waiting to pull an image for the first time.
+3. To test gitlab integration, add `-gitlab=true`
+4. All other tests.
+
+To run the tests:
```

Comment:
Hm. There is something with permissions when trying to run go tests in core-dev. when test calls kubectl it cannot access kubeconfig.

And also you cannot run it via docker container due to permissions:

Reply:
I'm sorry I missed your response from a couple days ago! :pray: 

I rebased with main and did ec61d396dfaf87dc3fd6f60cbda79c54b738ad78 to get the tests runnable in werft against core-dev. May I ask for you to review?

I just tried `kubectl apply` and had similar trouble in core-dev, which may be intentional (in hindsight) so that we do not "break" core-dev for other namespaces. In other words, `kubectl apply` is probably best for us (in hindsight) in single node clusters dedicated to ourselves (workspace preview, harvester, etc.)

Reply:
Looks good, thank you!